### PR TITLE
node: cpumanager: fix smtalign error message and minor cleanup

### PR DESCRIPTION
**Pull request by ffromani**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Fix SMTAlignmentError message if availablePhysicalCPUs = 0.  Without this change, the logic was mistakenly emitting     the old error message, which is confusing for troubleshooting. Also add a minor quality-of-life improvement for debuggability.

#### Which issue(s) this PR fixes:
Fixes N/A

#### Special notes for your reviewer:
Cosmetic only changes, but still relevant for debuggability. The error message was confusing and causing real pain on a corner case

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/cm/cpumanager/policy_static.go [hunk 1]
```diff
@@ -323,7 +328,7 @@ func (p *staticPolicy) Allocate(s state.State, pod *v1.Pod, container *v1.Contai
 			// CPUs on a physical core. Allocation of individual threads would never have to occur.
 			return SMTAlignmentError{
 				RequestedCPUs: numCPUs,
-				CpusPerCore:   CPUsPerCore,
+				CpusPerCore:   p.cpuGroupSize,
```
Comment (bart0sh) [inline pkg/kubelet/cm/cpumanager/policy_static.go:331]: nit:
```suggestion
				CpusPerCore:   p.cpuGroupSize,
				CausedByPhysicalCPUs:  false,
```
@ffromani up to you to consider or not. I found it a little bit more expressive and explicit.

**Comment by bart0sh** [comment]
/triage accepted
/lgtm

/assign @SergeyKanzhelev @mrunalp 
for approval

File: pkg/kubelet/cm/cpumanager/policy_static_test.go [hunk 2]
```diff
@@ -509,7 +509,7 @@ func TestStaticPolicyAdd(t *testing.T) {
 			stAssignments:   state.ContainerCPUAssignments{},
 			stDefaultCPUSet: cpuset.New(0, 2, 3, 4, 5, 7, 8, 9, 10, 11),
 			pod:             makePod("fakePod", "fakeContainerBug113537_1", "10000m", "10000m"),
-			expErr:          SMTAlignmentError{RequestedCPUs: 10, CpusPerCore: 2, AvailablePhysicalCPUs: 8},
+			expErr:          SMTAlignmentError{RequestedCPUs: 10, CpusPerCore: 2, AvailablePhysicalCPUs: 8, CausedByPhysicalCPUs: true},
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/cm/cpumanager/policy_static_test.go:512]: please add a test case for false that will demonstrate what was wrong

Comment (ffromani) [reply pkg/kubelet/cm/cpumanager/policy_static_test.go:512]: good point, added

File: pkg/kubelet/cm/cpumanager/policy_static.go (hunk 1 above)
Comment (ffromani) [reply pkg/kubelet/cm/cpumanager/policy_static.go:331]: added. Not sure it helps, but not against either so why not?

**Comment by SergeyKanzhelev** [comment]
please squash

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Pull request merged**